vendor/k8s.io/kubernetes/test/e2e/scheduling/preemption.go

```diff
@@ -95,17 +95,25 @@ var _ = SIGDescribe("SchedulerPreemption [Serial]", func() {
 		for i, node := range nodeList.Items {
 			currentCpuUsage, currentMemUsage := getCurrentPodUsageOnTheNode(node.Name, allPods.Items, podRequestedResource)
 			framework.Logf("Current cpu and memory usage %v, %v", currentCpuUsage, currentMemUsage)
-			currentNode, err := cs.CoreV1().Nodes().Get(node.Name, metav1.GetOptions{})
-			framework.ExpectNoError(err)
-			cpuAllocatable, found := currentNode.Status.Allocatable["cpu"]
+			cpuAllocatable, found := node.Status.Allocatable["cpu"]
 			Expect(found).To(Equal(true))
 			milliCPU := cpuAllocatable.MilliValue()
-			milliCPU = milliCPU * 40 / 100
-			memAllocatable, found := currentNode.Status.Allocatable["memory"]
+			// Just to be tolerant use 0.6 of resources available on the node
+			milliCPU = int64(float64(milliCPU-currentCpuUsage) * float64(0.6))
+			memAllocatable, found := node.Status.Allocatable["memory"]
 			Expect(found).To(Equal(true))
 			memory := memAllocatable.Value()
-			memory = memory * 60 / 100
+			// Just to be tolerant use 0.6 of resources available on the node
+			memory = int64(float64(memory-currentMemUsage) * float64(0.6))
 			podRes = v1.ResourceList{}
+			// Have a guard to ensure memory, cpu value is greater than 0. This can happen, if there are too many
+			// non-besteffort pods(We should have never satisfied these conditions because of the way test is setup)
+			if memory <= 0 {
+				memory = 0
+			}
+			if milliCPU <= 0 {
+				milliCPU = 0
+			}
```

Comment:
how will the low priority pod be preempted if it doesn't make any resource requests (i.e. runs best-effort)?

Reply:
As I have mentioned in the comment, we shouldn't enter this but if we're it means we're going past what can be allocatable out of the node.

Reply:
then why not just fail cleanly at this point rather than in some weird way that could be mistaken as a flake?

Reply:
That's true. The only thing I had in my mind back then was to ensure that tests won't fail because of -ve values being assigned to cpu and memory. I will remove it after the tests successfully pass.